map/traffic_manager.cpp

```diff
@@ -44,15 +49,40 @@ void TrafficManager::UpdateViewport(ScreenBase const & screen)
 
   // Request traffic.
   auto mwms = m_getMwmsByRectFn(screen.ClipRect());
-
-  // TODO: request new mwms, update old ones by timer.
-
   for (auto const & mwm : mwms)
-    RequestTrafficData(mwm);
+  {
+    if (mwm.IsAlive())
+    {
+      bool needRequesting = false;
+      {
+        lock_guard<mutex> lock(m_requestedMwmsLock);
+        auto it = m_requestTimings.find(mwm);
+        if (it == m_requestTimings.end())
+        {
+          needRequesting = true;
+          m_requestTimings[mwm] = steady_clock::now();
+        }
+        else
+        {
+          auto const passedSeconds = duration_cast<seconds>(steady_clock::now() - it->second);
```

Comment:
Do we need duration_cast here? Is it possible to compare any two time intervals?